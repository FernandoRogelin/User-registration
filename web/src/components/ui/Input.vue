<script setup>
import { Field, ErrorMessage } from 'vee-validate'

defineProps({
  name: {
    type: String,
    required: true
  },
  type: {
    type: String,
    default: ''
  },
  label: {
    type: String,
    default: ''
  }
})
</script>

<template>
  <div class="components-ui-input">
    <label>
      <span v-if="label" class="components-ui-input__label">{{ label }}</span>
      <Field :name="name" :type="type" :id="name" class="components-ui-input__field" />
      <ErrorMessage :name="name" as="div" class="components-ui-input__error" />
    </label>
  </div>
</template>

<style scoped lang="scss">
.components-ui-input {
  &__label {
    font-weight: 600;
    font-size: 0.8rem;
  }

  &__field {
    height: 20px;
    margin: 0.4rem 0 0;
    border-radius: 4px;
    border: 1px solid var(--mb-black);
  }

  &__error {
    font-weight: 500;
    font-size: 0.7rem;
    color: var(--mb-error);
  }
}
</style>
