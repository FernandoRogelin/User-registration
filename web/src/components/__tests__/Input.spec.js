import { describe, expect, test, afterEach } from "vitest"
import { render, screen, cleanup } from "@testing-library/vue"

import Input from "@/components/ui/Input.vue"

const mockProps = {
  name: 'email',
  type: 'text',
  label: 'Input component'
}

afterEach(() => cleanup())

describe('<Input /> ui component', () => {
  test('should render button component and match snapshot', () => {
    const { container } = render(Input, { props: mockProps })

    expect(container.firstChild).toMatchSnapshot()
  })

  test('should render text label', () => {
    render(Input, { props: mockProps })

    expect(screen.getByText(/Input component/i)).toBeInTheDocument()
  })
})
