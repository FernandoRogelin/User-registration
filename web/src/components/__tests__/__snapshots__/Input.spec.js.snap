// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Input /> ui component > should render button component and match snapshot 1`] = `
<div
  class="components-ui-input"
  data-v-fcd0ab8f=""
>
  <label
    class="components-ui-input-label"
    data-v-fcd0ab8f=""
  >
    <span
      class="components-ui-input-label__span"
      data-v-fcd0ab8f=""
    >
      Input component
    </span>
    <input
      class="components-ui-input-label__field"
      data-v-fcd0ab8f=""
      name="email"
      type="text"
    />
    <!---->
  </label>
</div>
`;
